gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -167,6 +183,20 @@ private static MapFeature toMapPin(Feature feature) {
         .build();
   }
 
+  private static MapFeature toMapPolygon(Feature feature) {
+    Type listType = new TypeToken<ArrayList<Point>>() {}.getType();
+    ArrayList<Point> points = new Gson().fromJson(feature.getPolygonVertices(), listType);
```

Comment:
If we're hiding the internal GeoJson representation, one would expect `feature.getPolygonVertices()` would return return `List<Point>`, and that additional GeoJson parsing wouldn't be needed. See discussion in first comment.